packages/mui-base/src/Slider/Root/useSliderRoot.ts

```diff
@@ -190,10 +188,31 @@ function useSliderRoot(parameters: UseSliderParameters): UseSliderReturnValue {
     [inputValidationRef],
   );
 
-  const { contextValue: compoundComponentContextValue, subitems } = useCompoundParent<
-    string,
-    SliderThumbMetadata
-  >();
+  const thumbRefs = React.useRef<(HTMLElement | null)[]>([]);
+
+  // Map with index (DOM position) as the key and the id attribute of each thumb <input> element as the value
+  const [inputIdMap, setInputMap] = React.useState(() => new Map<number, string>());
+
+  const deregisterInputId = React.useCallback((index: number) => {
+    setInputMap((prevMap) => {
+      const nextMap = new Map(prevMap);
+      nextMap.delete(index);
+      return nextMap;
+    });
+  }, []);
+
+  const registerInputId = React.useCallback(
+    (index: number, inputId: string | undefined) => {
+      if (index > -1 && inputId !== undefined) {
+        setInputMap((prevMap) => new Map(prevMap).set(index, inputId));
+      }
+
+      return {
+        deregister: deregisterInputId,
+      };
+    },
+    [deregisterInputId],
+  );
```

Comment:
Is it possible to modify the `Composite` source to allow for metadata instead?

Reply:
Should be! How about doing this when refactoring Tabs to use Composite? Tabs uses metadata more extensively, and I wanted to get this PR merged ASAP to unblock the ESM work @michaldudak is doing 🙏

Reply:
Seems ok to me as of now